supervisor/supervisor.go

```diff
@@ -275,8 +277,12 @@ func (s *Supervisor) Start() error {
 		"cpus":        s.machine.Cpus,
 	}).Debug("containerd: supervisor running")
 	go func() {
-		for i := range s.tasks {
-			s.handleTask(i)
+		for {
+			select {
+			case i := <-s.tasks:
```

Comment:
Something in that vein may be better. The current change remove the serialization of the command, since now they can happen in parallel which may lead to unplanned behavior.

wdyt @crosbymichael ? I'm a bit afraid of sync bug creeping up by removing the serialization.